libs/MeshKernel/include/MeshKernel/CurvilinearGrid/CurvilinearGridFromSplines.hpp

```diff
@@ -281,22 +284,22 @@ namespace meshkernel
-        std::vector<std::vector<double>> m_crossSplineCoordinates;               ///< t center spline coordinates of cross splines
-        std::vector<std::vector<double>> m_cosCrossingAngle;                     ///< cosPhi cosine of crossing angle
-        std::vector<std::vector<std::vector<double>>> m_crossSplineLeftHeights;  ///< hL left - hand side grid heights at cross spline locations for each grid layer subinterval, hL(1, :) being the height of the first subinterval, etc.
-        std::vector<std::vector<std::vector<double>>> m_crossSplineRightHeights; ///< hR right - hand side grid heights at cross spline locations for each grid layer subinterval, hR(1, :) being the height of the first subinterval, etc.
-        std::vector<std::vector<UInt>> m_numCrossSplineLeftHeights;              ///< NsubL number of subintervals of grid layers at cross spline locations at the left - hand side of the spline, each having their own exponential grow factor
-        std::vector<std::vector<UInt>> m_numCrossSplineRightHeights;             ///< NsubR number of subintervals of grid layers at cross spline locations at the right - hand side of the spline, each having their own exponential grow factor
-        std::vector<UInt> m_numMSplines;                                         ///< mfac number of grid intervals on the spline
-        std::vector<UInt> m_leftGridLineIndex;                                   ///< iL index in the whole gridline array of the first grid point on the left - hand side of the spline
-        std::vector<UInt> m_rightGridLineIndex;                                  ///< iR index in the whole gridline array of the first grid point on the right - hand side of the spline
-        std::vector<std::vector<double>> m_gridHeights;                          ///< Heights of all grid elements
+        std::vector<SplineTypes> m_type;                ///< Spline type
+        std::vector<int> m_centralSplineIndex;          ///< For each spline the index to its central
+        std::vector<UInt> m_numCrossingSplines;         ///< Number of crossing splines
+        lin_alg::Matrix<UInt> m_crossingSplinesIndices; ///< Indices for each cross spline, the indices of the center splines
+
+        lin_alg::Matrix<bool> m_isLeftOriented;                         ///< isLeftOriented cross spline is left to right(.true.) or not (.false.) w.r.t.center spline
+        lin_alg::Matrix<double> m_crossSplineCoordinates;               ///< t center spline coordinates of cross splines
+        lin_alg::Matrix<double> m_cosCrossingAngle;                     ///< cosPhi cosine of crossing angle
+        lin_alg::Matrix<std::vector<double>> m_crossSplineLeftHeights;  ///< hL left - hand side grid heights at cross spline locations for each grid layer subinterval, hL(1, :) being the height of the first subinterval, etc.
```

Comment:
What about Boost multi dimension arrays?

Reply:
It is an option.  The matrix  elements of type `lin_alg::Matrix<std::vector<double>>` are contiguous in memory. So I believe this is performant enough.